### ISSUE TITLE
kite#completion#popup_exit

### ISSUE THREAD

**Issue by craldaz** [kiteco/vim-plugin#326]
I had kite installed and it suddenly decided to give this error while using vim

`E117: Unknown function: kite#completion#popup_exit`

This  completely breaks vim because a new line cannot be made

I've uninstalled the vim plugin and kite but the problem persists. Attempting to re-install vim. Hopefully the problem doesn't persist. I just want to use vim again.

**Comment by craldaz** [comment]
Updating vim with brew didn't help. 

However, removing my .vimrc "fixed" the problem

here is my old .vimrc

```
map <F2> i#!/bin/bash<ESC>                                      
map <F3> o#This file was created on <ESC>:r!date "+\%x"<ESC>kJ  

au FileType c,cpp setlocal comments-=:// comments+=f://         
syntax on
:ab /**************************************** /****************************************
set incsearch
:ab #e ^V^H*****************************************/           
autocmd BufWinLeave *.* mkview                                  
autocmd BufWinEnter *.* silent loadview
highlight Comment ctermfg=Green                                 
" toggle between terminal and vim mouse                         
set mouse=n

filetype plugin indent on                                       
" show existing tab with 4 spaces width                         
set tabstop=4
" " when indenting with '>', use 4 spaces width                 
set shiftwidth=4                                                
" " On pressing tab, insert 4 spaces
set expandtab
inoremap <S-Tab> <C-d>

set visualbell
set t_vb=                                                       

augroup remember_folds                                          
  autocmd!                                                      
  autocmd BufWinLeave * mkview                                  
  autocmd BufWinEnter * silent! loadview                        
augroup END
```


I don't know what is causing the `kite#completion#popup_exit`

**Comment by craldaz** [comment]
Okay, clearing ~/.vim/views/ helped. Still not sure what's going on but vim works now

**Comment by airblade** [comment]
Thanks for posting your vimrc – that's really helpful.  These lines stood out to me:

```viml
autocmd BufWinLeave *.* mkview                                  
autocmd BufWinEnter *.* silent loadview
```

Vim's `:mkview` remembers mappings but not defined functions (see `:help mkview`).

The function `kite#completion#popup_exit()` was removed in e231c72, which was released in v1.0.76 of this plugin (you can get the version from `:echo kite#utils#plugin_version()`).

I think what happened was your kite plugin got updated from below v1.0.76 to v1.0.76 or above, and when you next entered a buffer, `:loadview` restored the mapping which pointed to `kite#completion#popup_exit()` but not the function itself.

There isn't much we can do about this – Vim's own docs for `:mkview` say that it's "not perfect" – apart from document it, which we'll do.  Still, I'm sorry you had this trouble.

If you wanted to re-install the Kite plugin it would work fine again.

**Comment by airblade** [comment]
Closing this now but please do open a new issue if you have any further problems.
